scaleway/resource_compute_instance_server.go

```diff
@@ -0,0 +1,360 @@
+				Computed:    true,
+				Description: "the Scaleway internal IP address of the server", // TODO: add this field in CreateServerRequest (proto)
+			},
+			"public_ip": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "the public IPv4 address of the server",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "started",
+				Description: "the state of the server should be: started, stopped, standby",
+				ValidateFunc: validation.StringInSlice([]string{
+					"started",
+					"stopped",
+					"standby",
+				}, false),
+			},
+			"user_data": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "the user data associated with the server", // TODO: document reserved keys (`cloud-init`)
+			},
+			"zone":       zoneSchema(),
+			"project_id": projectIDSchema(),
+		},
+	}
+}
+
+const giga = 1000000000
+
+func resourceScalewayComputeInstanceServerCreate(d *schema.ResourceData, m interface{}) error {
+	instanceApi, zone, err := getInstanceAPIWithZone(d, m)
+	if err != nil {
+		return err
+	}
+
+	name, ok := d.GetOk("name")
+	if !ok {
+		name = getRandomName("srv")
+	}
+	req := &instance.CreateServerRequest{
+		Zone:           zone,
+		Name:           name.(string),
+		Organization:   d.Get("project_id").(string),
+		Image:          d.Get("image").(string),
+		CommercialType: d.Get("type").(string),
+		EnableIPv6:     d.Get("enable_ipv6").(bool),
+		SecurityGroup:  d.Get("security_group_id").(string),
+	}
+
+	if raw, ok := d.GetOk("tags"); ok {
+		for _, tag := range raw.([]interface{}) {
+			req.Tags = append(req.Tags, tag.(string))
+		}
+	}
+
+	if size, ok := d.GetOk("root_volume.0.size_in_gb"); ok {
+		req.Volumes = make(map[string]*instance.VolumeTemplate)
+		req.Volumes["0"] = &instance.VolumeTemplate{
+			Size: uint64(size.(int)) * giga,
+		}
+	}
+
+	res, err := instanceApi.CreateServer(req)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newZonedId(zone, res.Server.ID))
+
+	// todo: add userdata
+
+	for _, action := range stateToAction("stopped", d.Get("state").(string)) {
+		err = instanceApi.ServerActionAndWait(&instance.ServerActionAndWaitRequest{
+			Zone:     zone,
+			ServerID: res.Server.ID,
+			Action:   action,
+			Timeout:  time.Minute * 10,
+		})
+		if err != nil && !is404Error(err) {
+			return err
+		}
+	}
+
+	return resourceScalewayComputeInstanceServerRead(d, m)
+}
+
+func resourceScalewayComputeInstanceServerRead(d *schema.ResourceData, m interface{}) error {
+	instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(m, d.Id())
+	if err != nil {
+		return err
+	}
+
+	response, err := instanceApi.GetServer(&instance.GetServerRequest{
+		Zone:     zone,
+		ServerID: ID,
+	})
+	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	switch response.Server.State {
+	case instance.ServerStateStopped:
+		d.Set("state", "stopped")
```

Comment:
Maybe we can create constants for the states? Because we use them on multiple places...